src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategy.kt

```diff
@@ -222,7 +245,9 @@ abstract class ManagedStrategy(
         runner.initialize()
         val loggedResults = runInvocation()
         val sameResultTypes = loggedResults.javaClass == failingResult.javaClass
-        val sameResults = loggedResults !is CompletedInvocationResult || failingResult !is CompletedInvocationResult || loggedResults.results == failingResult.results
+        val sameResults = loggedResults !is CompletedInvocationResult ||
+            failingResult !is CompletedInvocationResult ||
+            loggedResults.results.withoutCrashes == failingResult.results.withoutCrashes
```

Comment:
Should we not check that crashes are the same?

Reply:
The problem here is that during rerunning the failed interleaving the trace is collected, so crash exceptions are filled with stacktace info, so crashes in these two invocations always differ. I can disable stacktrace collection at all in MC mode as we have all trace collected, then crashes will contain no info => will be equal.

Reply:
Sorry, this is not the reason. I'll fix it.